### DISCUSSION TITLE
Fail if undeclared variables/functions have been used in the code

### DISCUSSION THREAD

**Discussion by dipakw** [babel/babel#14518] [Q&A]
Hi guys,

Is there any way to catch use of undeclared variables/functions? Any plugin etc?

For example, in the code below, I have used `divid`, an undeclared variable, it compiles successfully, is there any way to enable babel to catch those things?

```jsx
function MyComponent(){
  return(
    <div id={divid}>
       ...
    </div>
  )
}
```

**Comment by nicolo-ribaudo** [comment]
You could create a custom plugin to do it, but it looks like a linter's job (you could use ESlint, for example)

**Comment by dipakw** [comment]
Thanks mate, ESlint worked.
